mesonbuild/scripts/meson_exe.py

```diff
@@ -64,14 +65,31 @@ def run_exe(exe):
         child_env['PATH'] = (os.pathsep.join(exe.extra_paths + ['']) +
                              child_env['PATH'])
         if exe.exe_runner and mesonlib.substring_is_in_list('wine', exe.exe_runner.get_command()):
-            wine_paths = ['Z:' + p for p in exe.extra_paths]
-            wine_path = ';'.join(wine_paths)
-            # Don't accidentally end with an `;` because that will add the
-            # current directory and might cause unexpected behaviour
-            if 'WINEPATH' in child_env:
-                child_env['WINEPATH'] = wine_path + ';' + child_env['WINEPATH']
-            else:
-                child_env['WINEPATH'] = wine_path
+            seen = set()
+            wine_paths = ['Z:' + p for p in exe.extra_paths if not (p in seen or seen.add(p))]
+            wine_paths += [p for p in child_env.get('WINEPATH', '').split(';') if not (p in seen or seen.add(p))]
+
+            wine_shortpaths = []
```

Comment:
__[Flake8]__ [F841] local variable 'wine_shortpaths' is assigned to but never used ([link](https://sider.review/gh/repos/19784232/pulls/5756#issue-4405666))

<sub>posted by Sider</sub>